moto/sagemaker/models.py

```diff
@@ -0,0 +1,364 @@
+            raise RESTError(
+                error_type="ValidationException",
+                message=message,
+                template="error_json",
+            )
+
+    def validate_instance_type(self, instance_type):
+        VALID_INSTANCE_TYPES = [
+            "ml.p2.xlarge",
+            "ml.m5.4xlarge",
+            "ml.m4.16xlarge",
+            "ml.t3.xlarge",
+            "ml.p3.16xlarge",
+            "ml.t2.xlarge",
+            "ml.p2.16xlarge",
+            "ml.c4.2xlarge",
+            "ml.c5.2xlarge",
+            "ml.c4.4xlarge",
+            "ml.c5d.2xlarge",
+            "ml.c5.4xlarge",
+            "ml.c5d.4xlarge",
+            "ml.c4.8xlarge",
+            "ml.c5d.xlarge",
+            "ml.c5.9xlarge",
+            "ml.c5.xlarge",
+            "ml.c5d.9xlarge",
+            "ml.c4.xlarge",
+            "ml.t2.2xlarge",
+            "ml.c5d.18xlarge",
+            "ml.t3.2xlarge",
+            "ml.t3.medium",
+            "ml.t2.medium",
+            "ml.c5.18xlarge",
+            "ml.p3.2xlarge",
+            "ml.m5.xlarge",
+            "ml.m4.10xlarge",
+            "ml.t2.large",
+            "ml.m5.12xlarge",
+            "ml.m4.xlarge",
+            "ml.t3.large",
+            "ml.m5.24xlarge",
+            "ml.m4.2xlarge",
+            "ml.p2.8xlarge",
+            "ml.m5.2xlarge",
+            "ml.p3.8xlarge",
+            "ml.m4.4xlarge",
+        ]
+        if not validators.is_one_of(instance_type, VALID_INSTANCE_TYPES):
+            message = "Value '{}' at 'instanceType' failed to satisfy constraint: Member must satisfy enum value set: {}".format(
+                instance_type, VALID_INSTANCE_TYPES
+            )
+            raise RESTError(
+                error_type="ValidationException",
+                message=message,
+                template="error_json",
+            )
+
+    @property
+    def arn(self):
+        return (
+            "arn:aws:sagemaker:"
+            + self.region_name
+            + ":"
+            + str(ACCOUNT_ID)
+            + ":notebook-instance/"
+            + self.notebook_instance_name
+        )
+
+    @property
+    def url(self):
+        return "{}.notebook.{}.sagemaker.aws".format(
+            self.notebook_instance_name, self.region_name
+        )
+
+    def start(self):
+        self.status = "InService"
+
+    @property
+    def is_deletable(self):
+        return self.status in ["Stopped", "Failed"]
+
+    def stop(self):
+        self.status = "Stopped"
+
+
+class SageMakerModelBackend(BaseBackend):
+    def __init__(self, region_name=None):
+        self._models = {}
+        self.notebook_instances = {}
+        self.region_name = region_name
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+
+    def create_model(self, **kwargs):
+        model_obj = Model(
+            self.region_name,
+            kwargs.get("ModelName"),
```

Comment:
Use named variables for long argument lists like this - reduces the chance that we're passing in arguments in the wrong order

`Model(region_name=self.region_name, model_name=kwargs.get("ModelName"), ...)`

Reply:
Done.